File: Projects/PythonProject/Multithreading/ThreadLifeCycleExample.py
import threading
import time

class LifecycleDemo(threading.Thread):
    def run(self):
        print("Thread is in the Running state")
        time.sleep(0.5)  # Simulate some work

def execute_thread_lifecycle_example():
    thread = LifecycleDemo()
    print("Thread State: NEW (created but not started)")
    
    thread.start()
    print("Thread State: RUNNABLE (started, running or ready to run)")

    thread.join()  # Wait for thread to finish
    print("Thread State: TERMINATED (completed execution)")

if __name__ == "__main__":
    execute_thread_lifecycle_example()

File: Projects/JavaProject/src/com/Multithreading/ThreadLifeCycleExample.java
package com.Multithreading;

///Managing user sessions in a web application.

class LifecycleDemo implements Runnable {
    @Override
    public void run() {
        System.out.println("Thread is in the Running state");
    }
}

public class ThreadLifeCycleExample {
    public static void executeThreadLifecycleExample() throws InterruptedException {
        Thread thread = new Thread(new LifecycleDemo());
        System.out.println("Thread State: " + thread.getState()); // New
        thread.start();
        System.out.println("Thread State: " + thread.getState()); // Runnable
        thread.join(); // Wait for thread to finish
        System.out.println("Thread State: " + thread.getState()); // Terminated
    }

    public static void main(String[] args) throws InterruptedException {
        executeThreadLifecycleExample();
    }
}

